src/Workspaces/Remote/ServiceHub/Services/SemanticClassification/RemoteSemanticClassificationService.cs

```diff
@@ -36,7 +35,8 @@ public RemoteSemanticClassificationService(in ServiceConstructionArguments argum
                 using (UserOperationBooster.Boost())
                 {
                     var solution = await GetSolutionAsync(solutionInfo, cancellationToken).ConfigureAwait(false);
-                    var document = solution.GetDocument(documentId);
+                    var document = solution.GetDocument(documentId) ?? await solution.GetSourceGeneratedDocumentAsync(documentId, cancellationToken).ConfigureAwait(false);
```

Comment:
this pattern is in several places (and will become more common).  Can we have an appropriate extension: GetExitingOrSourceGeneratedDocumentAsync?

Reply:
I'll defer this work to the PR that does a pass of more feature changes like this so the patterns become more apparent.